lib/services/server.go

```diff
@@ -473,3 +475,30 @@ func NewAWSNodeFromEC2Instance(instance ec2Types.Instance, awsCloudMetadata *typ
 
 	return server, nil
 }
+
+// NewAWSNodeFromEC2v1Instance creates a Node resource from an EC2 Instance.
+// It has a pre-populated spec which contains info that is not available in the ec2.Instance object.
+// Uses AWS SDK Go V1
```

Comment:
Feels like we should default aws-sdk-go v2 for any new functionality. How hard would that be to do here?